src/routes/join/components/SignupPage.js

```diff
@@ -0,0 +1,180 @@
+const { default: Button } = require('@material-ui/core/Button');
+const { default: Box } = require('@material-ui/core/Box');
+const { default: Link } = require('@material-ui/core/Link');
+const { default: FormControl } = require('@material-ui/core/FormControl');
+const { default: InputLabel } = require('@material-ui/core/InputLabel');
+const { default: Input } = require('@material-ui/core/Input');
+const { default: InputAdornment } = require('@material-ui/core/InputAdornment');
+const { default: IconButton } = require('@material-ui/core/IconButton');
+const { default: Visibility } = require('@material-ui/icons/Visibility');
+const { default: VisibilityOff } = require('@material-ui/icons/VisibilityOff');
+const StrangeForms = require('strange-forms');
+
+const internals = {};
+
+module.exports = class SignupPage extends StrangeForms(React.Component) {
+
+    static propTypes = {
+        reqCreateAccount: T.func.isRequired
+    };
+
+    static fields = {
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: ''
+    };
+
+    defaultSubmit = (ev) => {
+
+        ev.preventDefault();
+        this.handleNext();
+    }
+
+    constructor(props) {
+
+        super(props);
+
+        this.state = {
+            firstName: '',
+            lastName: '',
+            email: '',
+            password: '',
+            showPassword: false,
+            isSubmitting: false
+        };
+
+        this.strangeForm({
+            fields: Object.keys(SignupPage.fields),
+            get: (someProps, field) => {
+
+                return someProps.data ? someProps.data[field] : '';
+            },
+            act: () => null
+        });
+    }
+
+    handleClickShowPassword = (ev) => {
+
+        this.setState((currentState) => ({
+            showPassword: !currentState.showPassword
+        }));
+    }
+
+    disableSubmit() {
+
+        return false;
+    }
+
+    handleSubmit = async (ev) => {
+
+        ev.preventDefault();
+        const accountInfo = this.formatFields();
+        this.setState({ isSubmitting: true });
+        const [err, result] = await this.props.reqCreateAccount(accountInfo);
+        this.setState({ isSubmitting: false });
+        if (!err && result.success) {
+            // Login and redirect
+        }
+    }
+
+    formatFields = () => {
+
+        return {
+            firstName: this.fieldValue('firstName'),
+            lastName: this.fieldValue('lastName'),
+            email: this.fieldValue('email'),
+            password: this.fieldValue('password'),
+            role: 'user'
+        };
+    };
+
+    render() {
+
+        const { isSubmitting, showPassword } = this.state;
+        const { HomepageContainer } = internals;
+
+        return (
+            <HomepageContainer>
+                <Typography variant='h4' align='center' gutterBottom>Sign Up</Typography>
+                <form
+                    style={{ display: 'flex', flexDirection: 'column' }}
```

Comment:
I don't have strong feelings, but this would be a place where we could use box as a drop-in replacement for inline styles: `<Box component='form' display='flex' flexDirection='column ...`.

Reply:
Oh snap, all these inline styles were meant to be temporary. I agree with this!